pydantic_core/_types.py

```diff
@@ -59,6 +59,7 @@ class DictSchema(TypedDict, total=False):
 
 class FloatSchema(TypedDict, total=False):
     type: Required[Literal['float']]
+    allow_inf_nan: bool  # whether 'NaN', '+inf', '-inf' should be forbidden. default: True
```

Comment:
You need to add `allow_inf_nan` to `Config` to otherwise it can't be set, might be good to add a test to confirm setting `allow_inf_nan` via config works.

Reply:
Ah yes I added the implementation but not the interfaces. I'll do that soon(ish)

Reply:
b0bf4adeb48b8ad7078fe33bc5b17a4f72ee2d03